src/main/telemetry/smartport.c

```diff
@@ -653,48 +653,73 @@ void handleSmartPortTelemetry(void)
                     uint32_t tmpi = 10000; // start off with at least one digit so the most significant 0 won't be cut off
 
                     // ones column
+                    // ARM states are either mutually exclusive or implied by an  other , 	room  to encode additional  mode states within the same ba
                     if (!isArmingDisabled())
-                        tmpi += 1;
-                    else
-                        tmpi += 2;
+                        tmpi = 1;    	// was tmpi += 1
+                    else		// meaning ready to ARM
+                        tmpi = 2;    	// was tmpi += 2
 
                     if (ARMING_FLAG(ARMED))
-                        tmpi += 4;
+                        tmpi = 3;      	// was  tmpi += 4
 
                     // tens column
+                    //   slot for mutually exclusive modes, allow to encode 10  modes within the available bandwidth
                     if (FLIGHT_MODE(ANGLE_MODE))
-                        tmpi += 10;
+                        tmpi = 10;
                     if (FLIGHT_MODE(HORIZON_MODE))
-                        tmpi += 20;
+                        tmpi = 20;
                     if (FLIGHT_MODE(PASSTHRU_MODE))
-                        tmpi += 40;
+                        tmpi = 30;
+                    if (FLIGHT_MODE(HEADING_MODE))  	// existing entry relocated here
+                        tmpi = 40;
+                    if (FLIGHT_MODE(NAV_LAUNCH_MODE))	// new entry post 1.7.3
+                        tmpi = 50;
+                    if (FLIGHT_MODE(NAV_RTH_MODE)) 		// existing entry relocated here
+                        tmpi = 60;
+                    if (FLIGHT_MODE(NAV_ALTHOLD_MODE)) 	// existing entry relocated here
+                        tmpi = 70;
+                    if (FLIGHT_MODE(NAV_POSHOLD_MODE)) 	// existing entry relocated here
```

Comment:
Please note that not all of these modes are mutially exclusive. POSHOLD & ALTHOLD can go together or independently. HEADING mode can be used together with ANGLE or HORIZON. ALTHOLD mode can be used together with ACRO, ANGLE or HORIZON as well.

Reply:
Yeah, this will simply not work. I'm willing to add fixed wing modes, and I offered to do so without breaking everything as this would do.

Reply:
This is not a problem of the enumeration process used  but of only of aggregating all of  those  so called  modifier modes  in separate slot (the hundred column) with other  mutually  exclusive modifier states still warranting the use of same efficient enumeration  of the 
 single digit tagging allowing 10 modes/sates  to be handled per slot

Reply:
Just move poshold  and  althold  to the  hundred columns  which would be the slot reserved for modifiers mode
No doubts the merits of the  efficient   straight 10 based  enumeration process  requires the proper aggregation of modes in the  5 available slots